main.py

```diff
@@ -103,16 +105,36 @@ def _create_flow_mod(version, packet, port):
         return flow_mod
 
     @staticmethod
-    def _create_packet_out(version, packet, ports):
+    def _create_packet_out(version, packet, ports, switch):
         """Create a PacketOut message with the appropriate version and data."""
         if version == '0x01':
-            packet_out = PacketOut10()
             port = ports[0] if ports else Port10.OFPP_FLOOD
+
+            try:
+                iface = switch.get_interface_by_port_no(port.value)
+            except AttributeError:
+                iface = switch.get_interface_by_port_no(port)
```

Comment:
My only concern is here, you are querying twice, in the worst-case scenario. IIUC, AttributeError is not raised by `get_interface_by_port_no`, so my suggestion is something like this:
```python
try:
   port_number = port.value
except AttributeError:
   port_number = port
iface = switch.get_interface_by_port_no(port_number)
```

Or we should handle this properly inside `get_interface_by_port_no`.